### PR TITLE
3.0.0

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: ProGetAutomation/Functions/Get-ProGetAssetContent.ps1 [hunk 1]
```diff
@@ -6,37 +6,47 @@ function Get-ProGetAssetContent
     Gets the content of an asset in an asset directory.
 
     .DESCRIPTION
-    The `Get-ProGetAssetContent` function gets an asset's content from ProGet. Pass the name of the root asset directory to the `DirectoryName` parameter. Pass the path to the asset to the `Path` parameter. If the URL to an asset directory in ProGet is `https://proget.example.com/assets/versions/subdirectory/file`, the directory parameter is the first directory after `assets/` (in this example `versions`). The `Path` parameter would be the rest of the url in this case `subdirectory/file`.
+    The `Get-ProGetAssetContent` function gets an asset's content from ProGet. Pass the name of the root asset directory
+    to the `DirectoryName` parameter. Pass the path to the asset to the `Path` parameter. If the URL to an asset
+    directory in ProGet is `https://proget.example.com/assets/versions/subdirectory/file`, the directory parameter is
+    the first directory after `assets/` (in this example `versions`). The `Path` parameter would be the rest of the url
```
Comment (decoyjoe) [inline ProGetAutomation/Functions/Get-ProGetAssetContent.ps1:12]: `the directory parameter` should be `the DirectoryName parameter`

File: ProGetAutomation/Functions/Get-ProGetAssetContent.ps1 [hunk 2]
```diff
@@ -6,37 +6,47 @@ function Get-ProGetAssetContent
...
     $uri = '/endpoints/{0}/content/{1}' -f $DirectoryName,$Path
 
-    return Invoke-ProGetRestMethod -Session $Session -Path $uri -Raw
+    $response = Invoke-ProGetRestMethod -Session $Session -Path $uri -Raw
+    if (-not $response)
+    {
+        return
+    }
+
+    return ,$response.Content
```
Comment (decoyjoe) [inline ProGetAutomation/Functions/Get-ProGetAssetContent.ps1:51]: I recently discovered that `Write-Output` has a `-NoEnumerate` switch. Does that behave the same? I.e. can we do:

```powershell
return $response.Content | Write-Output -NoEnumerate
```

File: ProGetAutomation/Functions/Invoke-ProGetRestMethod.ps1 [hunk 3]
```diff
@@ -32,13 +32,11 @@ function Invoke-ProGetRestMethod
         [Parameter(ParameterSetName='ByParameter')]
         [hashtable] $Parameter,
 
-        # Controls how the parameters are sent to the API. The default is `Form`, which sends them as URL-encoded
-        # name/value pairs (i.e. like a HTML form submission). The other options is `Json`, which converts the
-        # parameters to JSON and sends that JSON text as the content/body of the request. This parameter is ignored if
-        # there are no parmaeters to send or if the `InFile` parameter is used.
+        # Sends parameters to the API as JSON. By default, parameters are sent in the request body as URL-encoded
+        # name/value pairs (i.e. like a HTML form submission). This parameter is ignored if there are no parmaeters to
```
Comment (decoyjoe) [inline ProGetAutomation/Functions/Invoke-ProGetRestMethod.ps1:36]: Typo: `parmaeters`

File: ProGetAutomation/Functions/Set-ProGetAsset.ps1 [hunk 4]
```diff
@@ -53,7 +53,11 @@ function Set-ProGetAsset
 
         # The content to be published as an asset.
         [Parameter(Mandatory, ParameterSetName='ByContent')]
-        [String] $Content
+        [String] $Content,
+
+        # The asset's content type. By default, the asset's content type will be "application/" used to upload
+        # the file, which is most likely no what you want.
```
Comment (decoyjoe) [inline ProGetAutomation/Functions/Set-ProGetAsset.ps1:59]: Typos:
* `will be "application/" use to upload`
* `likely no what you want`

**Review by decoyjoe — APPROVED** [review]

File: ProGetAutomation/Functions/Get-ProGetAssetContent.ps1 (hunk 2 above)
Comment (splatteredbits) [reply ProGetAutomation/Functions/Get-ProGetAssetContent.ps1:51]: From the help: 

>    [!IMPORTANT] There is an issue with this switch in Windows PowerShell that is fixed in PowerShell 6.2 and 
above. When using NoEnumerate and explicitly using the InputObject parameter, the command > still enumerates. To work around this, pass the InputObject argument(s) positionally.

image

Comment (decoyjoe) [reply ProGetAutomation/Functions/Get-ProGetAssetContent.ps1:51]: 😞

**Pull request merged**